File: 백준/Bronze/9506. 약수들의 합/약수들의 합.py
while(1):
    n = int(input())
    if(n == -1): exit()
    hap = 0
    arr = []
    for i in range(1, n):
        if(n % i == 0): 
            arr.append(i)
            hap += i
        
    if(hap == n):
        print("{}".format(n), end=" = ")
        for i in range(len(arr) - 1):
            print("{}".format(arr[i]), end=" + ")
        print(arr[len(arr) -1])
    else:
        print("{} is NOT perfect.".format(n))

File: 백준/Bronze/9506. 약수들의 합/약수들의 합.c
#include <stdio.h>

int main(void) {
	int n;
	
	while (1) {
		scanf("%d", &n);

		if (n == -1) break;
		int arr[10000];
		int sum = 0, index = 0;
		for (int i = 1; i < n; i++) {
			if (n % i == 0) {
				sum += i;
				arr[index++] = i;
			}
		}
		if (sum == n) {
			printf("%d = %d", n, arr[0]);
			for (int i = 1; i < index; i++) {
				printf(" + %d", arr[i]);
			}
			printf("\n");
		}
		else {
			printf("%d is NOT perfect.\n", n);
		}
	}
	return 0; 
}